openmls/src/framing/codec.rs

```diff
@@ -210,3 +210,66 @@ impl tls_codec::Size for MlsMessage {
         }
     }
 }
+
+impl Size for MlsCiphertextContent {
+    fn tls_serialized_len(&self) -> usize {
+        self.content.tls_serialized_len() +
+           self.signature.tls_serialized_len() +
+            self.confirmation_tag.tls_serialized_len() +
+            // Note: The padding is appended as a "raw" all-zero byte slice
+            // with length `length_of_padding`. Thus, we only need to add
+            // this length here.
+            self.length_of_padding as usize
+    }
+}
+
+impl Serialize for MlsCiphertextContent {
+    fn tls_serialize<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+        let mut written = 0;
+
+        written += self.content.tls_serialize(writer)?;
+        written += self.signature.tls_serialize(writer)?;
+        written += self.confirmation_tag.tls_serialize(writer)?;
+        let padding = vec![0u8; self.length_of_padding as usize];
+        writer.write_all(&padding)?;
+        written += self.length_of_padding as usize;
+
+        Ok(written)
+    }
+}
+
+impl MlsCiphertextContent {
+    /// Deserialize an `MlsCiphertextContent` from a slice (with known length).
+    /// Note: We use a slice here and not `impl Read` because we need to infer the padding.
```

Comment:
I'm not sure I understand why you can't use `Read`. It provides a `read_to_end` function as well.

Reply:
From my understanding all other deserialization instances are expected to read as many bytes as needed to construct a specific object and leave the `Read` instance operational.

This is different here as this function kinda always "terminates" the `Read` instance due to `read_to_end`. A slice thus felt more appropriate (even though I use a cursor for technical reasons inside the function).

Furthermore, when we change that to `Read` the caller has to wrap a slice in a cursor.

Reply:
From what I see you add complexity where it's not needed by using something else. Why do you think you need a cursor?

Reply:
> Why do you think you need a cursor?

Nevermind. No need for a `Cursor` here.

Reply:
I refactored to use `tls_codec::Deserialize` and added a note regarding termination of the `Read` instance.